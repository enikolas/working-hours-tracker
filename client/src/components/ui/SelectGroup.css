.SelectGroup {
  min-height: 40px;
}

.SelectGroup .label {
  width: 60px;
  margin-right: 10px;
  font-size: 11px;
  float: left;
  text-align: right;
  clear: both;
  height: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.SelectGroup select {
  border: 2px solid #909fa8;
  width: 230px;
  font-size: 13px;
  padding: 4px;
  background-color: #fff;
  margin-bottom: 20px;
}

.SelectGroup div {
  font-size: 14px;
  height: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
