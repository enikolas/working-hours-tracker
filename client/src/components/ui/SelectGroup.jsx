import React from 'react';
import PropTypes from 'prop-types';

import './SelectGroup.css';

const renderOptions = options => options.map(option => (
  <option key={option} value={option}>
    {option}
  </option>
));

const SelectGroup = ({
  label,
  options,
  selected,
  onChange,
  showTextInstead,
  isDisabled
}) => (
  <fieldset className="SelectGroup">
    <label>
      <span className="label">{ label }</span>
      { showTextInstead ?
        <div>
          { showTextInstead }
        </div> :
        <select
          value={selected || ''}
          onChange={event => onChange(event.target.value)}
          disabled={isDisabled}
        >
          { renderOptions(options) }
        </select>
      }
    </label>
  </fieldset>
);

SelectGroup.propTypes = {
  label: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string),
  selected: PropTypes.string,
  onChange: PropTypes.func,
  showTextInstead: PropTypes.string,
  isDisabled: PropTypes.bool
};

SelectGroup.defaultProps = {
  selected: '',
  options: [],
  onChange: null,
  showTextInstead: null,
  isDisabled: false
};

export default SelectGroup;
