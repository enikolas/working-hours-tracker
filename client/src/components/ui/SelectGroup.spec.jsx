import React from 'react';
import { shallow } from 'enzyme';

import SelectGroup from './SelectGroup';

const mockOptions = ['one', 'two', 'three'];

describe('SelectGroup', () => {
	describe('render', () => {
		it('should show a select component with min props', () => {
			const wrapper = shallow(<SelectGroup name="testName" label="testLabel" />);
			expect(wrapper).toMatchSnapshot();
		});
		it('should call onChange with selected option', () => {
			const onChangeFn = jest.fn();
			const wrapper = shallow(<SelectGroup
				name="testName"
				label="testLabel"
				options={mockOptions}
				onChange={onChangeFn}
			/>);
			const selectField = wrapper.find('select');
			selectField.simulate('change', { target: { value: 2 } });
			expect(onChangeFn).toHaveBeenCalledWith(2);
		});
	});
});
