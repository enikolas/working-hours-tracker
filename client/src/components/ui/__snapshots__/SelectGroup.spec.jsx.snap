// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGroup render should show a select component with min props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectGroup
    isDisabled={false}
    label="testLabel"
    name="testName"
    onChange={null}
    options={Array []}
    selected=""
    showTextInstead={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label>
        <span
          className="label"
        >
          testLabel
        </span>
        <select
          disabled={false}
          onChange={[Function]}
          value=""
        />
      </label>,
      "className": "SelectGroup",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="label"
          >
            testLabel
          </span>,
          <select
            disabled={false}
            onChange={[Function]}
            value=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "testLabel",
            "className": "label",
          },
          "ref": null,
          "rendered": "testLabel",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "disabled": false,
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": Array [],
          "type": "select",
        },
      ],
      "type": "label",
    },
    "type": "fieldset",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label>
          <span
            className="label"
          >
            testLabel
          </span>
          <select
            disabled={false}
            onChange={[Function]}
            value=""
          />
        </label>,
        "className": "SelectGroup",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="label"
            >
              testLabel
            </span>,
            <select
              disabled={false}
              onChange={[Function]}
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testLabel",
              "className": "label",
            },
            "ref": null,
            "rendered": "testLabel",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "disabled": false,
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Array [],
            "type": "select",
          },
        ],
        "type": "label",
      },
      "type": "fieldset",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
